state/secrets.go

```diff
@@ -877,21 +899,32 @@ func (st *State) checkConsumerCountOps(uri *secrets.URI, inc int) ([]txn.Op, err
 }
 
 func secretOwnerLabelKey(ownerTag string, label string) string {
-	return fmt.Sprintf("secretlabel#%s#%s", ownerTag, label)
+	return fmt.Sprintf("secretOwnerlabel#%s#%s", ownerTag, label)
+}
+
+func secretConsumerLabelKey(ownerTag string, label string) string {
+	return fmt.Sprintf("secretConsumerlabel#%s#%s", ownerTag, label)
 }
 
-func (st *State) uniqueSecretLabelOps(ownerTag string, label string) ([]txn.Op, error) {
+func (st *State) uniqueSecretOwnerLabelOps(ownerTag string, label string) ([]txn.Op, error) {
+	return st.uniqueSecretLabelOps(ownerTag, label, secretOwnerLabelKey)
+}
+
+func (st *State) uniqueSecretConsumerLabelOps(consumerTag string, label string) ([]txn.Op, error) {
+	return st.uniqueSecretLabelOps(consumerTag, label, secretConsumerLabelKey)
+}
+
+func (st *State) uniqueSecretLabelOps(tag string, label string, keyGenerator func(string, string) string) ([]txn.Op, error) {
 	refCountCollection, ccloser := st.db().GetCollection(refcountsC)
 	defer ccloser()
 
-	key := secretOwnerLabelKey(ownerTag, label)
+	key := keyGenerator(tag, label)
 	countOp, count, err := nsRefcounts.CurrentOp(refCountCollection, key)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	if count > 0 {
-		return nil, errors.WithType(
-			errors.Errorf("secret label %q owned by %s already exists", label, ownerTag), LabelExists)
+		return nil, errors.WithType(errors.Errorf("label %q for %q already exists", label, tag), LabelExists)
```

Comment:
we should retain "secret" in the message. also, can we add "owner" or "consumer", ie pass in a badge param
`errors.Errorf("secret %s label %q for %q already exists", badge, label, tag)`